lua/nvim_lsp/codeqlls.lua

```diff
@@ -0,0 +1,46 @@
+local configs = require "nvim_lsp/configs"
+local util = require "nvim_lsp/util"
+
+local server_name = "codeqlls"
+
+local root_pattern = util.root_pattern("qlpack.yml")
+
+configs[server_name] = {
+    default_config = {
+        filetypes = {'codeql'};
+        root_dir = function(fname)
+            return root_pattern("qlpack.yml") or util.path.dirname(fname)
+        end;
+        log_level = vim.lsp.protocol.MessageType.Warning;
+        before_init = function(initialize_params, config)
+            initialize_params['workspaceFolders'] = {{
+                    name = 'workspace',
+                    uri = initialize_params['rootUri']
+            }}
+        end;
+        settings = {
+            search_path = vim.empty_dict()
```

Comment:
This should be `{}`. `vim.empty_dic()` acts as empty dictionary.
But I think you want search_path to be `list`.

Reply:
For whatever reason, using `{}` breaks it. But you are right, it should be an empty list? is there any similar `vim.empty_list()`

Reply:
What breaks mean?
Would you show me something like error messages?

Reply:
> is there any similar `vim.empty_list()`

no, that would be identical to `{}`

Reply:
But some reason, print statements do no not work within the `on_new_config` callback (or in any other callback as far as I know), so Im not able to debug why `search_path` is not accounted for when usin `{}` instead of an empty dict.

I have a simple codeql query like:

```
import java
import foo
```

If `search_path` is not accounted for, both lines will contain errors. If `search_path` is correctly passed to the language server, only `import foo` will get an error.

So using an empty dict, I get the expected error, but when using an empty list I get also an error on `import java` meaning that search path was not updated.

Is there a way to debug callbacks? 
Thanks

Reply:
I was able to see what was happening by logging to a file rather than using `print`. So when I use `vim.empty_dict()`, `settings.search_path` from the user config overrides the default config. But if I use `{}` instead, `config.settings` at the beggining of the `on_new_config` callback is `{ ["search_path"] = { } ,} `

So it seems like when using `{}`, user config do not override default config

Reply:
So apparently the bug is un `nvim_lsp/util.tbl_deep_extend`:

```
dst[k] = v
```

`v` comes from default config and `dst` is the user config. In that line, the user config is always overriden with default config, no matter the contents.

It can be reproduced with:

```
function test()
    user_config = { ["search_path"] = { "~/codeql-home/codeql-repo"}}
    print(dump(user_config))
    default_config = { ["search_path"] = { } ,}
    print(dump(default_config))
    new_config = util.tbl_deep_extend(user_config, default_config)
    print(dump(new_config))
end
```

Reply:
Added a new commit that fixes the issue for me, but not sure if something else is needed (eg: extend lists?)

Reply:
May be related with #98